sympy/solvers/guesser.py

```diff
@@ -0,0 +1,81 @@
+"""
+Formula guessing
+================
+
+Given a list of expressions generated by an unknown function over
+an integer parameter
+
+.. math::
+
+    a_{0\le n\le N}=\{a_0, a_1, \ldots, a_N\},
+
+it is possible to guess recurrent relations between the values, and
+further to deduce an expression producing the given values
+(a "closed form"). At the moment SymPy can guess linear relations,
+leading to linear recurrences, from which all expressions of type
+
+.. math::
+
+    p_1\cdot c_1^n + p_2\cdot c_2^n + \cdots
+
+(`p_i` polynomial in `n`, `c_i` constant expression) can be guessed,
+provided enough values are given. Such sequences of values are called
+"C-finite".
+
+References
+----------
+
+.. [1] Zeilberger, Doron. "The C-finite ansatz." The Ramanujan
+    Journal 31.1-2 (2013): 23-32.
+"""
+
+def guess(seq):
+    """
+    Return closed form depending on `n` that produces the sequence,
```

Comment:
Why not remove out trivial steps, involving rsolve and return instead a recurrence?